phoenix-core/src/main/java/org/apache/phoenix/index/GlobalIndexChecker.java

```diff
@@ -202,7 +208,42 @@ private void init() throws IOException {
             else {
                 pageSize = Long.MAX_VALUE;
             }
+
+            Filter filter = scan.getFilter();
+            if (filter instanceof PagingFilter) {
```

Comment:
We may also need to wrap non paging filters. So the same logic should be applied even when the scan filter is not an instance of PagingFilter.